specification/purview/data-plane/Azure.Analytics.Purview.PDS/preview/2022-11-01-preview/pds.json

```diff
@@ -0,0 +1,463 @@
+            "in": "query",
+            "name": "isProjectGondor",
+            "description": "Is call from a Gondor client.",
+            "type": "boolean"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DatasourcePreferencesResponse"
+            }
+          },
+          "default": {
+            "description": "An unexpected error occurred",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DatasourcePreferences_Get": {
+            "$ref": "./examples/DatasourcePreferences_Get.json"
+          }
+        }
+      }
+    },
+    "/policy-elements": {
+      "get": {
+        "tags": [
+          "PolicyElements"
+        ],
+        "description": "Get policies for given scope.",
+        "operationId": "PolicyElements_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "scope",
+            "description": "Scope.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "The API Version.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "filter",
+            "description": "filter.",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PolicyResponse"
+            }
+          },
+          "default": {
+            "description": "An unexpected error occurred",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "values"
+        },
+        "x-ms-examples": {
+          "PolicyElements_Get": {
+            "$ref": "./examples/PolicyElements_Get.json"
+          }
+        }
+      }
+    },
+    "/policy-events": {
+      "get": {
+        "tags": [
+          "PolicyEvents"
+        ],
+        "description": "Get policy events for given scope.",
+        "operationId": "PolicyEvents_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "scope",
+            "description": "Scope.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "syncToken",
+            "description": "The sync token.",
+            "required": true,
```

Comment:
Is this parameter really required? If so, where does the user get a value to supply in this parameter?  I realize that there will be a syncToken in the response, and I use that on a subsequent request, but what about the first request?

Reply:
Hi Mike, as we discussed earlier, we are following the sync token approach for returning the delta changes.
The initial value of sync token is obtained from "policyElements" API response (Please check the "PolicyResponse" model). This will internally call as full pull.
Once the user has this initial value, they pass this value to subsequent "policyEvents" calls (we call this as delta pull).
In delta pull we return the delta changes to the input resource if there are any. Along with this information we return the most recent value for sync token.
Clients persist the latest value of this sync token, so that they can call delta API requesting for newer changes.
And suppose there is a failure on client side due to any reason, the cycle can be restarted again by calling the full pull.
I hope I have explained the importance of sync token in the delta pull (policyEvents) API.
I am resolving the conversation based on this explanation.
Let me know if this needs further clarification.

Reply:
Okay I remember now. But please add these details to the descriptions in the REST API itself, so that users will understand how to use it.